nixos/modules/services/continuous-integration/gitlab-runner-multi.nix

```
@@ -0,0 +1,231 @@
+            type = types.listOf types.str;
+            default = [ ];
+            description = ''
+              Custom environment variables injected to build environment.
+            '';
+          };
+          executor = mkOption {
+            type = types.str;
+            default = "docker";
+            description = ''
+              Select executor, eg. shell, docker, etc.
+            '';
+          };
+          dockerImage = mkOption {
+            type = types.nullOr types.str;
+            default = null;
+            description = ''
+              Docker image to be used.
+            '';
+          };
+          dockerVolumes = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            description = ''
+              Bind-mount a volume and create it if it doesn't exist prior to mounting.
+            '';
+          };
+          dockerDisableCache = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Disable all container caching.
+            '';
+          };
+          preCloneScript = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            description = ''
+              Runner-specific command script executed before code is pulled.
+            '';
+          };
+          preBuildScript = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            description = ''
+              Runner-specific command script executed after code is pulled, just before build executes.
+            '';
+          };
+          postBuildScript = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            description = ''
+              Runner-specific command script executed after code is pulled and just after build executes.
+            '';
+          };
+          tagList = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            description = ''
+              Tag list.
+            '';
+          };
+          runUntagged = mkOption {
+            default = false;
+            type = types.bool;
+            description = ''
+              Register to run untagged builds;
+              defaults to 'true' when 'tag-list' is empty.
+            '';
+          };
+          limit = mkOption {
+            default = 0;
+            type = types.int;
+            description = ''
+              Limit how many jobs can be handled concurrently by this service.
+              0 (default) simply means don’t limit
+            '';
+          };
+          requestConcurrency = mkOption {
+            default = 0;
+            type = types.int;
+            description = ''
+              Limit number of concurrent requests for new jobs from GitLab.
+            '';
+          };
+        };
+      });
+    };
+  };
+  config = mkIf (cfg.enable && cfg.services != { }) {
```

```suggestion
  config = mkIf cfg.enable {
```

Shouldn't this be like that?

---

There is no point in running `gitlab-runner` without any runner services. It also will complain that  there is no `/var/lib/gitlab-runner/.gitlab-runner/config.toml`.